tap-snapshots/test-arborist-reify.js-TAP.test.js

```diff
@@ -13521,23 +13521,55 @@ Node {
     Node {
       "children": Map {
         "@ruyadorno/package-with-added-bin" => Node {
+          "edgesIn": Set {
+            Edge {
+              "from": "a",
+              "name": "@ruyadorno/package-with-added-bin",
+              "spec": "^1.0.0",
+              "type": "prod",
+            },
+          },
           "location": "a/node_modules/@ruyadorno/package-with-added-bin",
           "name": "@ruyadorno/package-with-added-bin",
           "resolved": "https://registry.npmjs.org/@ruyadorno/package-with-added-bin/-/package-with-added-bin-1.0.0.tgz",
         },
       },
+      "edgesOut": Map {
+        "@ruyadorno/package-with-added-bin" => Edge {
+          "name": "@ruyadorno/package-with-added-bin",
+          "spec": "^1.0.0",
+          "to": "a/node_modules/@ruyadorno/package-with-added-bin",
+          "type": "prod",
+        },
+      },
       "location": "a",
       "name": "a",
       "resolved": "a",
     },
     Node {
       "children": Map {
         "@ruyadorno/package-with-added-bin" => Node {
+          "edgesIn": Set {
+            Edge {
+              "from": "b",
+              "name": "@ruyadorno/package-with-added-bin",
+              "spec": "^2.0.0",
+              "type": "prod",
+            },
+          },
           "location": "b/node_modules/@ruyadorno/package-with-added-bin",
           "name": "@ruyadorno/package-with-added-bin",
           "resolved": "https://registry.npmjs.org/@ruyadorno/package-with-added-bin/-/package-with-added-bin-2.0.0.tgz",
         },
       },
+      "edgesOut": Map {
+        "@ruyadorno/package-with-added-bin" => Edge {
+          "name": "@ruyadorno/package-with-added-bin",
+          "spec": "^2.0.0",
+          "to": "b/node_modules/@ruyadorno/package-with-added-bin",
+          "type": "prod",
+        },
+      },
```

Comment:
why are these new `edgesIn` and `edgesOut` getting added here?

Reply:
Because I had to make `loadVirtual` read the package.json on linked packages, or else their dependencies don't get loaded, and we can't calculate the audit report properly.

This is sort of a deeper version of the fix we made in buildIdealTree to fix the issue you found earlier with bin files.  Those edges _should_ be there, since they're a legitimate dependency in the graph, but weren't found, because the legacy shrinkwraps don't store dependency relationships for link targets.

Reply:
oohh makes sense! 👍